app/controllers/concerns/v2/bulk_assign.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module V2
+  # Logic for bulk assigning entities
+  module BulkAssign
+    extend ActiveSupport::Concern
+
+    included do
+      def bulk_assign(entity, relation_type)
+        return unless (input_list = permitted_params[:relationship]&.dig(relation_type))
+
+        entity.update_relationships(input_list, current_user)
+      end
+    end
+  end
```

Comment:
I'm not sure if this needs to be in a separate concern and I guess the `return` isn't even necessary, the "validation" can be moved into the model from the controller.